packages/reports/src/reports/blocks/types/renderers/Selector.jsx

```diff
@@ -18,17 +18,29 @@ export default function SelectorPreview({id, config}) {
 
   const [value, setValue] = useState(query[`selector${id}id`] || config.defaultValue);
   const [multiValue, setMultiValue] = useState(query[`selector${id}id`] ? query[`selector${id}id`].split(",") : config.defaultValue);
+  const [data, setData] = useState(config.options?.map(d => ({value: d.id, label: d.label})) || []);
+  const [serializedOptions, setSerializedOptions] = useState((config.options || []).map(d => d.id).join("-"));
 
-  const onChangeSingle = e => {
-    dispatch(setStatus({query: {[`selector${id}id`]: e}}));
-    setValue(e);
-  };
+  // change value if option list changes
+  useEffect(() => {
+    // options change ref when setStatus is called, so encode option content to compare equality 
+    const serializedNew = config.options.map(d => d.id).join("-");
```

Comment:
Even though its tiny, we should extract out this "serializer" so it's not repeated in two places